crates/ntx-builder/src/builder/mod.rs

```diff
@@ -0,0 +1,96 @@
+use std::{net::SocketAddr, time::Duration};
+
+use anyhow::Context;
+use miden_node_proto::generated::ntx_builder::api_server;
+use server::NtxBuilderApi;
+use store::StoreClient;
+use tokio::{net::TcpListener, time};
+use tokio_stream::wrappers::TcpListenerStream;
+use tower_http::trace::TraceLayer;
+use tracing::{debug, info, warn};
+
+use crate::COMPONENT;
+
+mod server;
+
+/// Interval for checking pending notes
+const NOTE_CHECK_INTERVAL_MS: u64 = 100;
+
+//mod data_store;
+mod store;
+
+/// Network Transaction Builder
+pub struct NetworkTransactionBuilder {
+    /// The address of the network transaction builder gRPC server.
+    pub address: SocketAddr,
+    /// Address of the store gRPC server.
+    pub store_address: SocketAddr,
```

Comment:
Irrelevant to this PR but is this an example where `SocketAddr` is the wrong abstraction because it means we can't use domains, only IPs? I have seen this in a few parts of the code base and made me wonder.

Reply:
Yeah, I didn't think too much about this when writing the code but I agree it'd probably be better to comprehensively use `Url`